### PR TITLE
Make Collect methods const

### PR DESCRIPTION
**Pull request by BenjaminW3**
Collect methods should not influence the state of the objects in any
externally visible way.

### CODE REVIEW

File: core/src/family.cc [hunk 1]
```diff
@@ -69,7 +69,7 @@ const std::map<std::string, std::string> Family<T>::GetConstantLabels() const {
 }
 
 template <typename T>
-std::vector<MetricFamily> Family<T>::Collect() {
+std::vector<MetricFamily> Family<T>::Collect() cont {
```
Comment (jupp0r) [inline core/src/family.cc]: typo here

Comment (BenjaminW3) [reply core/src/family.cc]: fixed

File: core/include/prometheus/collectable.h [hunk 2]
```diff
@@ -19,7 +19,7 @@ class PROMETHEUS_CPP_CORE_EXPORT Collectable {
   virtual ~Collectable() = default;
 
   /// \brief Returns a list of metrics and their samples.
-  virtual std::vector<MetricFamily> Collect() = 0;
+  virtual std::vector<MetricFamily> Collect() const = 0;
```
Comment (gjasny) [inline core/include/prometheus/collectable.h]: Changing the exposed ABI demands a version bump in root CMakLists.txt to 0.9.0

**Comment by gjasny** [comment]
Build could be fixed by appending a `@v1` to `uses: actions/checkout`

**Comment by BenjaminW3** [comment]
Thanks for fixing the internal mutability issues!
The coverage build still fails.

**Pull request merged**